client.go

```diff
@@ -387,6 +386,8 @@ func (c *client) reconnect() {
 	go alllogic(c)
 	go outgoing(c)
 	go incoming(c)
+
+	c.resume(false)
```

Comment:
Maybe I have misunderstood the code, but it seems weird that resume() returns a set of tokens, but there's no opportunity to wait on them.  I don't know how that might best be fixed.  Maybe the tokens could be stored in the client object and available for retrieval via some function call?

Reply:
I confess I wrote this patch quickly. I needed this for a project with a very short deadline. And as it is, the resume function should not return anything as the array of token is never used. Moreover I was not sure of the wanted behaviour for this library. Should I write a function to wait for all token being completed ? Let the user access this array of token and let him deal with it ? Call a user defined callback when all tokens are completed ? Or removes this array of token ?

I hope my explanations are clear. English is not my native language

Reply:
I think those are great questions.  If I have a chance I will go survey the other MQTT libraries and see what their strategy is.

Reply:
The idea behind the tokens was to allow you do effectively do synchronous operations with an async messaging system, so you can do a send of a QoS2 message and block until the message is full sent and acknowledged by the server. If we've been interrupted and are reloading this data from the Store the original tokens have already been closed and there is no easy way to associate new tokens with the original actions. If you can't think of a good reason or way to pass this back to the user I'm happy for now to just swallow them inside the library.

Reply:
Ok, juste let me removes the tokens array, if it is not used there is no need to allocate a eventually 2^16 length array.